js/messaging.js

```diff
@@ -233,12 +238,42 @@ function openURL({
       url.replace(/%2F.*/, '*').replace(/#.*/, '') :
       url.replace(/#.*/, '');
 
-  return queryTabs({url: urlQuery, currentWindow}).then(maybeSwitch);
+  return manageMatch || editMatch ? queryTabs().then(maybeSwitch) :
+  queryTabs({url: urlQuery, currentWindow}).then(maybeSwitch);
 
   function maybeSwitch(tabs = []) {
     const urlWithSlash = url + '/';
     const urlFF = FIREFOX && url.replace(/%2F/g, '/');
-    const tab = tabs.find(({url: u}) => u === url || u === urlFF || u === urlWithSlash);
+    const urlOptions = manageMatch ? URLS.ownOrigin + 'manage.html#stylus-options' : null;
+    const urlManage = manageMatch ? URLS.ownOrigin + 'manage.html' : null;
+    const tab = tabs.find(({url: u}) => u === url || u === urlFF || u === urlWithSlash ||
+                          u === urlOptions || u === urlManage);
+    if (!tab && prefs.get('openEditInWindow') && chrome.windows && editMatch) {
+      chrome.windows.create(
+        Object.assign({
+          url: url
+        }, prefs.get('windowPosition', {}))
+      );
+      return;
+    }
+    if (manageMatch) {
+      if (tab) {
+        const toggleOptions = url === urlOptions ? 'options-open' : 'options-close';
+        chrome.tabs.sendMessage(tab.id, {
+          'name': 'options',
+          'data': toggleOptions
+        });
+      }
+      getActiveTab()
```

Comment:
It checks if the existing tab is in another window, so it can send a message to close the popup if it's not a different FF window. Focusing existing tabs in other Windows in FF closes the popup on its own, but if you use `window.close()` it interferes with tab activation.